src/layers/domain/core/product_team/v3.py

```diff
@@ -22,9 +31,23 @@ class ProductTeam(AggregateRoot):
     ProductTeams, meaning that `ods_code` is not unique amongst ProductTeams.
     """
 
-    id: UUID
+    id: str = None
     name: str = Field(regex=ENTITY_NAME_REGEX)
     ods_code: str
+    status: Status = Status.ACTIVE
+    created_on: datetime = Field(default_factory=now, immutable=True)
+    updated_on: datetime = Field(default=None)
+    deleted_on: datetime = Field(default=None)
+    keys: list[ProductTeamKey] = Field(default_factory=list)
+
+    @root_validator(pre=True)
+    def set_id(cls, values):
+        if values.get("id") is None:
+            ods_code = values.get("ods_code")
+            if ods_code:
+                product_team_id = ProductTeamId.create(ods_code=ods_code)
+                values["id"] = product_team_id.id
```

Comment:
maybe just to tidy up the indenting:

```suggestion
    def set_id(cls, values):
        ods_code = values.get("ods_code")
        if ods_code and not values.get("id"):
            values["id"] = ProductTeamId.create(ods_code=ods_code)
```